design-doc-template.adoc

```diff
@@ -4,125 +4,95 @@ categories:
 
-* [ ] Preview
-
-* [ ] Community
-
-* [ ] default
-
-=== Dev Contacts
-
-* mailto:{email}[{author}]
+== Overview
 
-=== QE Contacts
+__<Define the requirement here. Be clear and succinct, you should be able to clearly define the context or problem in two or three paragraphs (if not sentences). Try to define the problem in the overall context and not to get into too much technical detail at this point.>__
 
-=== Testing By
-// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
-// Discuss with QE during the Kickoff state to decide this
-* [ ] Engineering
 
-* [ ] QE
+== Affected Projects or Components
 
-=== Affected Projects or Components
+__<List the projects or components that are affected by the feature. List them using their Git repositories.>__
 
 === Other Interested Projects
 
 === Relevant Installation Types
-// Remove the x next to the relevant field if the feature in question is not relevant
-// to that kind of WildFly installation
-* [x] Traditional standalone server (unzipped or provisioned by Galleon)
-
-* [x] Managed domain
 
-* [x] OpenShift s2i
+__<List the installation  types thar are relevant for the features and remove any that is not relevant>__.
 
-* [x] Bootable jar
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* Managed domain
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
 
 == Requirements
 
-=== Hard Requirements
-
-=== Nice-to-Have Requirements
-// Requirements in this section do not have to be met to merge the proposed functionality.
-// Note: Nice-to-have requirements that don't end up being implemented as part of
-// the work covered by this proposal should be moved to the 'Future Work' section.
-
+__<Describe the requirements that must be fullfilled by this feature.>__
 
 === Non-Requirements
-// Use this section to explicitly discuss things that readers might think are required
-// but which are not required.
+
+__<Use this section to explicitly discuss things that readers might think are required but which are not required.>__ 
 
 === Future Work
-// Use this section to discuss requirements that are not addressed by this proposal
-// but which may be addressed in later proposals.
+
+__<Use this section to discuss requirements that are not addressed by this proposal but which may be addressed in later proposals.>__
 
 == Backwards Compatibility
 
-// Does this enhancement affect backwards compatibility with previously released
-// versions of WildFly?
-// Can the identified incompatibility be avoided?
+__<Does this enhancement affect backwards compatibility with previously released versions of WildFly? Can the identified incompatibility be avoided?>__
 
 === Default Configuration
 
+__<What is the impact of this feature in the default configuration(s) provided by WildFly?>__
```

Comment:
Note that this and the following items are specific areas to consider in terms incompatibility. They are not top level items for general description.

I'd like to have a discussion somewhere about the exact meaning of this one and the "Importing Existing Configuration" one as there are relationships. And they both need to be more fully described. For example we provide *standard* OOTB configurations, while configuration items, used or not in those standard configs, have default values. This one should primarily be about the latter, but there are various cases to consider.

Reply:
I think this is what this should say:

__<Does this feature change the default value of any current configuration attributes? Does it change the configuration generated by any current Galleon layers?>__